django/db/models/query.py

```diff
@@ -460,6 +505,21 @@ def iterator(self, chunk_size=None):
         )
         return self._iterator(use_chunked_fetch, chunk_size)
 
+    async def aiterator(self, chunk_size=2000):
+        """
+        An asynchronous iterator over the results from applying this QuerySet
+        to the database.
+        """
+        if chunk_size <= 0:
+            raise ValueError("Chunk size must be strictly positive.")
+        use_chunked_fetch = not connections[self.db].settings_dict.get(
+            "DISABLE_SERVER_SIDE_CURSORS"
+        )
+        async for item in self._iterable_class(
+            self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size
+        ):
+            yield item
```

Comment:
What about `prefetch_related()`? It's a new method so we should raise `ValueError` when `aiterator()` is used after `prefetch_related()`, e.g.
```python
async def aiterator(self, chunk_size=2000): 
    if chunk_size is None:
        if self._prefetch_related_lookups:
            raise ValueError(
                "chunk_size must be provided when using QuerySet.iterator() after "
                "prefetch_related()."
            )
    elif chunk_size <= 0:
    ....
```